### PR TITLE
prov/efa: Minor cleanups

### PR DESCRIPTION
**Pull request by wckzhang**
Minor cleanups patch.

### CODE REVIEW

File: prov/efa/src/efa.h [hunk 1]
```diff
@@ -469,6 +469,10 @@ static inline bool efa_ep_is_cuda_mr(struct efa_mr *efa_mr)
 	return efa_mr ? (efa_mr->peer.iface == FI_HMEM_CUDA): false;
 }
 
+/*
```
Comment (bwbarrett) [inline prov/efa/src/efa.h:472]: using the format that the team has been using already...

File: prov/efa/src/efa_domain.c [hunk 2]
```diff
@@ -325,11 +325,16 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 			EFA_INFO(FI_LOG_DOMAIN, "EFA MR cache enabled, max_cnt: %zu max_size: %zu\n",
 			         cache_params.max_cnt, cache_params.max_size);
 			return 0;
+		} else {
```
Comment (bwbarrett) [inline prov/efa/src/efa_domain.c:328]: This is not what I meant.  I meant that the common case end of function should be obvious before you get into cleanup.  There's no need to jump into disable_caches if we never allocated a cache.  So make that an error state, and return 0 right before that.  THen remove the return 0 in line 327.

**Review by bwbarrett — CHANGES_REQUESTED** [review]

**Review by bwbarrett — APPROVED** [review]

**Comment by wckzhang** [comment]
aws:bot:retest

**Comment by shefty** [comment]
Is the world happy with this PR?

**Comment by rwespetal** [comment]
aws:bot:retest

**Review by rwespetal — APPROVED** [review]

**Comment by rwespetal** [comment]
aws:bot:retest

**Pull request merged**